gluten-iceberg/src/main/scala/org/apache/iceberg/spark/source/GlutenIcebergSourceUtil.scala

```diff
@@ -45,7 +47,7 @@ object GlutenIcebergSourceUtil {
           paths.add(task.file().path().toString)
           starts.add(task.start())
           lengths.add(task.length())
-          partitionColumns.add(new JHashMap[String, String]())
+          partitionColumns.add(getPartitionColumns(task))
```

Comment:
Can we assume the tasks share same `PartitionColumns` so that we don't need to get it for each task?

Reply:
The partition column is generally the same, but the partition values may vary.
<img width="529" alt="image" src="https://github.com/oap-project/gluten/assets/13622031/1c43e684-b8c3-4d02-899a-3e49bd6e406d">

Reply:
Get it. Thanks for show case.